deployment/SampleClaimsProvider/Controllers/V20210519/ClaimsProviderController.cs

```diff
@@ -0,0 +1,136 @@
+        /// <summary> This API retrieves claims for a specified vehicleId.
+        /// <example>
+        /// [GET] https://{hostname}/api/claims/claimInfo/vehicles/testVehicle1?paths=path1
+        /// </example>
+        /// </summary>
+        [HttpGet]
+        [Route("vehicles/{vehicleId}")]
+        public async Task<IActionResult> RetrieveClaimsWithVehicleId(string vehicleId)
+        {
+            return await this.RetrieveClaimsAsync(this.Request, vehicleId, userId: null);
+        }
+
+        /// <summary> This API retrieves claims for a specified vehicleId and userId.
+        /// <example>
+        /// [GET] https://{hostname}/api/claims/claimInfo/vehicles/testVehicle1/users/testUser1?paths=path1
+        /// </example>
+        /// </summary>
+        [HttpGet]
+        [Route("vehicles/{vehicleId}/users/{userId}")]
+        public async Task<IActionResult> RetrieveClaimsWithVehicleAndUserId(string vehicleId, string userId)
+        {
+            return await this.RetrieveClaimsAsync(this.Request, vehicleId, userId);
+        }
+
+        /// <summary> This API retrieves claims for a specified vehicleId.
+        /// <example>
+        /// [GET] https://{hostname}/api/claims/claimInfo/vehicles/testVehicle1?paths=path1
+        /// [Body] {"Path1": ["label1","label2"]} 
+        /// </example>
+        /// </summary>
+        [HttpPost]
+        [Route("vehicles/{vehicleId}/labels")]
+        public async Task<IActionResult> RetrieveClaimsByLabel(string vehicleId)
+        {
+            using StreamReader reader = new StreamReader(this.Request.Body);
+            string bodyContent = await reader.ReadToEndAsync();
+
+            Dictionary<string, IList<string>> labelsAndPaths = JsonConvert.DeserializeObject<Dictionary<string, IList<string>>>(bodyContent);
+            return await this.RetrieveLabeledClaimsAsync(this.Request, vehicleId, labelsAndPaths);
+        }
+
+        private static string GetAuthTokenIfExists(HttpRequest req)
+        {
+            return req.Headers.TryGetValue(Headers.AuthToken, out StringValues headerValue) ? headerValue.ToString() : string.Empty;
+        }
+
+        private async Task<IActionResult> RetrieveLabeledClaimsAsync(
+            HttpRequest req,
+            string vehicleId,
+            Dictionary<string, IList<string>> labelsAndPaths)
+        {
+            this.logger.LogInformation("[{tag}] Processing RetrieveLabeledClaimsAsync for vehicle with vehicleId '{vehicleId}'", "82b54258-e700-42d1-92de-b45aec151e98", vehicleId);
+
+            string authToken = GetAuthTokenIfExists(req);
+
+            LabeledClaimsResult claims = new LabeledClaimsResult
+            {
+                VehicleId = vehicleId,
+                UserId = null,
+                ExpiryTime = DateTime.UtcNow.AddDays(7),
+                LabeledClaims = new Dictionary<string, Dictionary<string, Collection<string>>>()
+            };
+
+            foreach ((string label, IList<string> paths) in labelsAndPaths)
```

Comment:
As a note, this logic here is why we don't need to deal with labeled claims in the `ClaimsProviderStore`